src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -455,19 +455,18 @@ private TypeSymbol BindTupleType(TupleTypeSyntax syntax, DiagnosticBag diagnosti
 
             if (hasExplicitNames)
             {
-                // If the tuple type with names is bound in a declaration
-                // context then we must have the TupleElementNamesAttribute to emit
-                if (syntax.IsTypeInContextWhichNeedsTupleNamesAttribute())
-                {
-                    // Report diagnostics if System.String doesn't exist
-                    this.GetSpecialType(SpecialType.System_String, diagnostics, syntax);
+                // If the tuple type with names is bound we must have the TupleElementNamesAttribute to emit
+                // it is typically there though, if we have ValueTuple at all
+                // and we need System.String as well
 
-                    if (!Compilation.HasTupleNamesAttributes)
-                    {
-                        var info = new CSDiagnosticInfo(ErrorCode.ERR_TupleElementNamesAttributeMissing,
-                            AttributeDescription.TupleElementNamesAttribute.FullName);
-                        Error(diagnostics, info, syntax);
-                    }
+                // Report diagnostics if System.String doesn't exist
+                this.GetSpecialType(SpecialType.System_String, diagnostics, syntax);
+
+                if (!Compilation.HasTupleNamesAttributes)
```

Comment:
I can see as well a possibility that the rule is relaxed to only cover cases where attribute is emitted. I will enter a tracking bug to finalize the decision.

---
In reply to: [80977532](https://github.com/dotnet/roslyn/pull/14023#discussion_r80977532) [](ancestors = 80977532,80973604)